client/src/api/block_builder/input_selection/new/requirement/base_token.rs

```diff
@@ -0,0 +1,160 @@
+    let inputs_sum = selected_inputs.iter().map(|input| input.output.amount()).sum::<u64>();
+    let outputs_sum = outputs.iter().map(|output| output.output.amount()).sum::<u64>();
+
+    (inputs_sum, outputs_sum)
+}
+
+impl InputSelection {
+    pub(crate) fn fulfill_base_token_requirement(&mut self) -> Result<(Vec<InputSigningData>, Option<Requirement>)> {
+        let (mut inputs_sum, mut outputs_sum) = base_token_sums(&self.selected_inputs, &self.outputs);
+        let mut newly_selected_inputs = Vec::new();
+        let mut newly_selected_ids = HashSet::new();
+
+        if inputs_sum >= outputs_sum {
+            return Ok((newly_selected_inputs, None));
+        }
+
+        // TODO don't pick burned things
+
+        // TODO if consolidate strategy: sum all the lowest amount until diff is covered.
+        // TODO this would be lowest amount of input strategy.
+        self.available_inputs
+            .sort_by(|left, right| left.output.amount().cmp(&right.output.amount()));
+
+        'overall: {
+            // 1. Basic with ED25519 address and nothing else
+            {
+                let inputs = self.available_inputs.iter().filter(|input| {
+                    if let Output::Basic(output) = &input.output {
+                        if let [UnlockCondition::Address(address)] = output.unlock_conditions().as_ref() {
+                            address.address().is_ed25519()
+                        } else {
+                            false
+                        }
+                    } else {
+                        false
+                    }
+                });
+
+                for input in inputs {
+                    inputs_sum += input.output.amount();
+                    newly_selected_inputs.push(input.clone());
+                    newly_selected_ids.insert(*input.output_id());
+
+                    if inputs_sum >= outputs_sum {
+                        break 'overall;
+                    }
+                }
+            }
+
+            // 2. Basic with ED25519 address and other things ????
+            {
+                // TODO
+            }
+
+            // 3. Moving funds of already transitioned other outputs
+            {
+                // TODO
+            }
+
+            // 4. Basic with other kind of address
+            {
+                // TODO
+            }
+
+            // 5. Other kinds of outputs
+            {
+                let mut inputs = self
+                    .available_inputs
+                    .iter()
+                    .filter(|input| !input.output.is_basic())
+                    .peekable();
+
+                if inputs.peek().is_some() {
+                    for input in inputs {
+                        inputs_sum += input.output.amount();
+                        newly_selected_inputs.push(input.clone());
+                        newly_selected_ids.insert(*input.output_id());
+
+                        if inputs_sum >= outputs_sum {
```

Comment:
This would be wrong if there is an input that requires us to create a storage deposit return output?

Reply:
Fixed